app/views/users/show.html.erb

```diff
@@ -1,5 +1,41 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %></h2>
+  <% if @user != current_user %>
+    <%= display_add_friend(current_user, @user)%>
+  <% end %>
+  
+
+  <br />
+  <p>Total number of friends: <%= @user.friendships.length %></p>
+
+  <h3>List of approved friends</h3>
+  <% @user.friends.each_with_index do |friend, index| friend %>
+    <p>
+      <%= index+1 %> - <%= friend.name %>
+      <% if @user == current_user %>
+        <%= display_add_friend(current_user, friend)%>
+      <% end %>
+    </p>
+  <% end %>
+
+  <h3>List of pending friends</h3>
+  <% @user.friend_requests.each do |p_friend| %>
+    <%= p_friend.name %>
+    <% if @user == current_user %>
+      <%=link_to 'Approve',  approve_friendship_path(user_id: @user.id, friend_id: p_friend.id)%>
+      
+      <%= link_to 'Decline', decline_friendship_path(user_id: @user.id, friend_id: p_friend.id)%>
+      
+    <% end %>
+  <% end %>
+
+  <h3>List of friends requests sent</h3>
+  <% @user.pending_friends.each_with_index do |friend, index| friend %>
+    <p><%= index+1 %> - <%= friend.name %> Friend Request sent</p>
+    
+  <% end %>
```

Comment:
## Remove Logic from this view

- [ ] Let's make sure that we have NO application logic in any of our Views nor helpers. We can encapsulate our view related logic in View helpers (app/helpers/\*_helper.rb) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright.